test/integration/video/assets.spec.js

```diff
@@ -155,4 +155,40 @@ describe('Integration::Assets', () => {
         err => expect(err).to.exist
       ));
   });
+
+  /** @test {Assets.updateMp4Support} */
+  describe('Assets.updateMp4Support', () => {
+    /** @test {Assets.updateMp4Support} */
+    it('updates the mp4 support for an asset', async () => {
+      try {
+        await Video.Assets.updateMp4Support(testAsset.id, {
+          mp4_support: 'standard',
+        });
+        const { mp4_support: updatedMp4Support } = await Video.Assets.get(
+          testAsset.id
+        );
+        expect(updatedMp4Support).to.equal('standard');
+      } catch (err) {
+        expect(err.messages && err.messages[0]).to.equal('Asset is not ready');
```

Comment:
So... ya... you can't update mp4 support before the asset is ready so these integration tests are less-than ideal.

The good news is that even when the asset is *not ready* the api endpoints will still return a 400 error if the params are incorrect. For example if the request is with params `{mp4_support: "blahblah"}` it would return a 400 error. So this integration test _at least_ verifies valid params are being sent

Alternative: In this integration test: poll and wait for the asset to be ready, then try to change the mp4_support...

Thoughts?

Reply:
I think this is ok- I really rely on these integration tests to make sure the endpoint is actually being hit and it sounds like it is.